flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/operators/StreamSourceOperatorTest.java

```diff
@@ -224,6 +226,39 @@ public void testLatencyMarkEmission() throws Exception {
 		Assert.assertTrue(output.get(i).isWatermark());
 	}
 
+	@Test
+	public void testManualWatermarkContextWatermarkMetric() throws Exception {
```

Comment:
Probably should add verifying the metric value after emitting the `Long.MAX_VALUE` watermark to this test, just to be on the safe side.